libraries/azurerm_resource.rb

```diff
@@ -5,42 +5,36 @@
 class AzurermResource < Inspec.resource(1)
   supports platform: 'azure'
 
-  MANAGEMENT_HOST = 'https://management.azure.com'
-  GRAPH_HOST      = 'https://graph.windows.net'
-
-  def client
+  def management
     Azure::Management.instance
-                     .with_client(rest_client)
+                     .with_client(management_client)
                      .for_subscription(subscription_id)
   end
 
-  def graph_client
+  def graph
+    raise Inspec::Exceptions::ResourceSkipped, 'MSI Authentication is currently unsupported '\
+      'for Graph RBAC API, control skipped.' if inspec.backend.msi_auth?
```

Comment:
Maybe:
```ruby
if inspec.backend.msi_auth?
  raise Inspec::Exceptions::ResourceSkipped, 'MSI Authentication is currently unsupported for Active Directory Users.'
end
```
Doing a guard clause puts the conditional at the front, which is the most important thing here. I reworded the error message so a user would have an easier time connecting that the resource isn't supported at the moment without needed to know what the Graph API is.

Reply:
Putting this in a conditional structure would definitely make the condition prominent, but why not extract the conditional behavior to a method with an explaining name, to meet both interests?

```ruby
def graph
  skip_msi_unsupported if inspec.backend.msi_auth?

  # ...
end

def skip_msi_unsupported
  raise Inspec::Exceptions::ResourceSkipped,
        'MSI Authentication is currently unsupported for Active Directory Users.'
end
```